src/main.rs

```diff
@@ -0,0 +1,139 @@
+//! # Network Error Logging
+//!
+//! A Compute@Edge service which exposes a HTTP reporting endpoint for the
+//! W3C [Network Error Logging API][specification].
+//!
+//! [specification]: https://www.w3.org/TR/network-error-logging
+use chrono::Utc;
+use fastly::http::{header, Method, StatusCode};
+use fastly::log::Endpoint;
+use fastly::{downstream_client_ip_addr, Body, Error, Request, Response, ResponseExt};
+use serde::{Deserialize, Serialize};
+use std::io::Write;
+
+// Import the report data structures.
+mod report;
+use report::Report;
+
+mod client_data;
+use client_data::ClientData;
```

Comment:
I might move our `mod`'s up above the `use` statements, and then import these with something like:

```
use crate::report::Report;
```

The `crate::` prefix, while optional, I think is somewhat helpful for disambiguating between imports coming from external dependencies and imports that come from within the project.